src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_BinaryOperator.cs

```diff
@@ -297,10 +299,10 @@ public BoundExpression VisitBinaryOperator(BoundBinaryOperator node, BoundUnaryO
                         if (loweredRight.ConstantValue == ConstantValue.True) return loweredLeft;
 
                         if (loweredLeft.ConstantValue == ConstantValue.False)
-                            return MakeUnaryOperator(UnaryOperatorKind.BoolLogicalNegation, syntax, null, loweredRight, loweredRight.Type);
+                            return MakeUnaryOperator(UnaryOperatorKind.BoolLogicalNegation, syntax, null, loweredRight, loweredRight.Type!);
```

Comment:
How do we know? (same below)

Reply:
Because it was classified as a bool equal (see `case`).  I'll assert.

---
In reply to: [386599433](https://github.com/dotnet/roslyn/pull/42068#discussion_r386599433) [](ancestors = 386599433)